### PR TITLE
feat: enhance CtwaReferralSource model with org field and backfill functionality

### PR DESCRIPTION
**Pull request by paulobernardoaf**
## Changes

- Added a `ForeignKey` field for `org` in the `CtwaReferralSource` model to associate referral sources with organizations.
- Updated the unique constraint on `CtwaReferralSource` to include `org`, allowing the same `source_id` and `source_type` combination to exist across different organizations.
- Added a composite index on `org` and `last_seen_at` to support efficient per-org queries.
- Introduced a `get_or_create_for_org` class method on `CtwaReferralSource` that enforces org is always provided when creating referral sources.
- Implemented a backfill migration that resolves the org for each existing `CtwaReferralSource` by tracing its associated CTWA channels, cloning records and retargeting CTWAs when a source is shared across multiple organizations.
- Made the `org` field non-nullable after backfilling to enforce data integrity going forward.
- Added tests covering model-level constraints, the `get_or_create_for_org` method, backfill logic including shared-org cloning, orphaned source logging, and edge cases such as empty inputs.

### CODE REVIEW

**Comment by paulobernardoaf** [comment]
* **#856** <a href="https://app.graphite.com/github/pr/weni-ai/flows/856?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/weni-ai/flows/856?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: temba/conversion_events/migrations/0004_backfill_ctwareferralsource_org.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                "source_url": source.source_url,
+                "headline": source.headline,
+                "body": source.body,
+            },
+        )
+
+        channel_uuids = list(Channel.objects.filter(org_id=extra_org_id).values_list("uuid", flat=True))
+        CTWA.objects.filter(referral_source_id=source.id, channel_uuid__in=channel_uuids).update(
+            referral_source_id=clone.id
+        )
```
[the remark this answers is not in the document]
Comment (paulobernardoaf) [reply temba/conversion_events/migrations/0004_backfill_ctwareferralsource_org.py]: The batch loop at line 79 is already the transaction boundary and nothing inside it catches exceptions, so any failure rolls back the whole batch and `max_id` is never advanced. A nested `atomic()` would only add a per-source savepoint that can never be used for partial recovery. `apply_orgs_to_source` is migration-local with a single production caller (`forwards`); the "tests call it directly" argument is not a production risk.

**Review by MatheusLeall — APPROVED** [review]

**Review by MatheusLeall — APPROVED** [review]

**Comment by paulobernardoaf** [comment]
### Merge activity

* **Aug 18, 8:50 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/weni-ai/flows/856).
* **Aug 18, 8:50 PM UTC**: [Graphite](https://app.graphite.com/github/pr/weni-ai/flows/856) couldn't merge this PR because it failed for an unknown reason (GitHub reports that this branch cannot be rebased. This is typically caused by merge conflicts or branch state issues. Please resolve any conflicts and retry the merge.).
* **Aug 18, 8:51 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/weni-ai/flows/856).
* **Aug 18, 8:51 PM UTC**: @paulobernardoaf merged this pull request with [Graphite](https://app.graphite.com/github/pr/weni-ai/flows/856).
